File: src/main/java/github/lms/approval/application/ApprovalResult.java
package github.lms.approval.application;

import github.lms.approval.domain.ApprovalStatus;
import github.lms.approval.domain.StepStatus;

import java.util.UUID;

public record ApprovalResult(
        UUID approvalId,
        ApprovalStatus approvalStatus,
        Long version,
        UUID activeStepId,
        StepStatus activeStepStatus,
        Integer activeStepOrder
) {}

File: src/main/kotlin/github/lms/approval/application/ApprovalResult.kt
package github.lms.approval.application

import github.lms.approval.domain.ApprovalStatus
import github.lms.approval.domain.StepStatus
import java.util.*

data class ApprovalResult(
    val approvalId: UUID,
    val approvalStatus: ApprovalStatus,
    val version: Long?,
    val activeStepId: UUID?,
    val activeStepStatus: StepStatus?,
    val activeStepOrder: Int?
)
